### PR TITLE
Remove gotos from ride.c

### PR DESCRIPTION
**Pull request by Gymnasiast**
Sorry for the formatting changes, it took enough effort to properly understand the code flow as it was. Besides the formatting changes, I have deliberately kept the other changes to a minimum.

### CODE REVIEW

File: src/openrct2/ride/ride.c [hunk 1]
```diff
@@ -3450,153 +3490,157 @@ void ride_set_map_tooltip(rct_map_element *mapElement)
...
+            break;
+        case 2:
+            rotatedCoords.x = x - y;
+            rotatedCoords.y = ((-y - x) / 2) - z;
+            break;
+        case 3:
+            rotatedCoords.x = y + x;
+            rotatedCoords.y = ((x - y) / 2) - z;
+            break;
         }
```
Comment (IntelOrca) [inline src/openrct2/ride/ride.c]: The same switch block can be found in `window_ride_construction_draw_track_piece`, so you could extract this to a new function in map for both usages.

Comment (Gymnasiast) [reply src/openrct2/ride/ride.c]: It's not entirely the same kind of calculation, by the look of it.

Comment (IntelOrca) [reply src/openrct2/ride/ride.c]: https://github.com/OpenRCT2/OpenRCT2/blob/5c98c7ee5eefacffd4df03f57680955021fb608b/src/openrct2/windows/RideConstruction.cpp#L2286-2304

Comment (Gymnasiast) [reply src/openrct2/ride/ride.c]: Ok, I split it off.

**Comment by janisozaur** [comment]
I'd still like to inspect it closer.

File: src/openrct2/ride/ride.c [hunk 2]
```diff
@@ -6970,26 +7056,34 @@ static sint32 ride_get_track_length(Ride *ride)
             if (mapElement->base_height != z)
                 continue;
 
-            goto foundTrack;
-        } while (!map_element_is_last_for_tile(mapElement++));
+            foundTrack = true;
+        }
+        while (!map_element_is_last_for_tile(mapElement++) && !foundTrack);
```
Comment (janisozaur) [inline src/openrct2/ride/ride.c]: This is not correct. Previous code would skip over incrementing `mapElement` if a track was found. `mapElement` is used later on, so it needs to be fixed. Change the order of conditions here to make it match.

File: src/openrct2/ride/ride.c [hunk 3]
```diff
@@ -6942,14 +7025,16 @@ static sint32 ride_get_smallest_station_length(Ride *ride)
  *
  *  rct2: 0x006CB3AA
  */
-static sint32 ride_get_track_length(Ride *ride)
+static sint32 ride_get_track_length(Ride * ride)
 {
-    rct_window *w;
-    rct_map_element *mapElement;
+    rct_window             * w;
+    rct_map_element        * mapElement = NULL;
```
Comment (janisozaur) [inline src/openrct2/ride/ride.c]: minor alignment issues

File: src/openrct2/ride/ride.c [hunk 4]
```diff
@@ -6530,40 +6578,49 @@ static sint32 loc_6CD18E(sint16 mapX, sint16 mapY, sint16 entranceMinX, sint16 e
...
 {
-    sint16 mapX, mapY;
-    sint16 entranceMinX, entranceMinY, entranceMaxX, entranceMaxY, word_F4418C, word_F4418E;
-    sint32 interactionType, direction, stationHeight, stationDirection;
-    rct_map_element *mapElement;
-    rct_viewport *viewport;
-    Ride *ride;
+    sint16          mapX, mapY;
+    sint16          entranceMinX, entranceMinY, entranceMaxX, entranceMaxY, word_F4418C, word_F4418E;
+    sint32          interactionType, direction, stationHeight, stationDirection;
```
Comment (janisozaur) [inline src/openrct2/ride/ride.c]: misaligned wrt pointers below

**Pull request merged**